### PR TITLE
ffmpeg: misc cleanups

### PR DESCRIPTION
**Pull request by pbogdan**
###### Motivation for this change

There are 5 versions of ffmpeg included in nixpkgs, 2 of which are not supported upstream as I understand it. This attempts to switch packages to a more up to date version where possible. As a result `ffmpeg_1` can be dropped; `ffmpeg_0` still remains :/ with a single reverse dependency `gst-ffmpeg`. The packages depending on `gst-ffmpeg` are:

- [x] `miro` - depends on other deps marked as insecure already - `webkitgtk24x-gtk2` - so perhaps could be dropped? - https://github.com/NixOS/nixpkgs/pull/50354
- [x] `gnash` - as per https://github.com/NixOS/nixpkgs/pull/26620 gstreamer support is not enabled by default and some parts of it don't work well? Could the gstreamer support be removed here? - https://github.com/NixOS/nixpkgs/pull/50481
- [x] `transcribe` - I think it should be possible to update the package and switch to new gstreamer and drop the `gst-ffmpeg` dependency - https://github.com/NixOS/nixpkgs/pull/50337
- [x] `tor-browser-bundle{-bin}` - don't know what to do with those.. - https://github.com/NixOS/nixpkgs/pull/50373

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

I verified things build on `x86_64-linux` but not tested much beyond that.

### CODE REVIEW

**Comment by jtojnar** [comment]
Related issues: https://github.com/NixOS/nixpkgs/issues/22372, https://github.com/NixOS/nixpkgs/issues/15930

-----

:+1: for dropping Miro, for https://github.com/NixOS/nixpkgs/issues/39975

File: pkgs/tools/audio/acoustid-fingerprinter/default.nix [hunk 1]
```diff
@@ -15,10 +15,13 @@ stdenv.mkDerivation rec {
...
-  patches = [ (fetchpatch {
-    url = "https://bitbucket.org/acoustid/acoustid-fingerprinter/commits/632e87969c3a5562a5d4842b03613267ba6236b2/raw";
-    sha256 = "15hm9knrpqn3yqrwyjz4zh2aypwbcycd0c5svrsy1fb2h2rh05jk";
-  }) ];
+  patches = [
+    (fetchpatch {
+      url = "https://bitbucket.org/acoustid/acoustid-fingerprinter/commits/632e87969c3a5562a5d4842b03613267ba6236b2/raw";
+      sha256 = "15hm9knrpqn3yqrwyjz4zh2aypwbcycd0c5svrsy1fb2h2rh05jk";
+    })
+    ./ffmpeg.patch
```
Comment (jtojnar) [inline pkgs/tools/audio/acoustid-fingerprinter/default.nix]: Could you add source of the path in a comment?

File: pkgs/applications/audio/spotify/default.nix [hunk 2]
```diff
@@ -23,7 +23,7 @@ let
     curl
     dbus
     expat
-    ffmpeg_0_10
+    ffmpeg
```
Comment (jtojnar) [inline pkgs/applications/audio/spotify/default.nix]: Does this really work (can local files be played)? Arch package requires https://aur.archlinux.org/packages/ffmpeg-compat-54/

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -9525,9 +9525,6 @@ with pkgs;
   ffmpeg_0_10 = callPackage ../development/libraries/ffmpeg/0.10.nix {
     inherit (darwin.apple_sdk.frameworks) Cocoa;
   };
-  ffmpeg_1_2 = callPackage ../development/libraries/ffmpeg/1.2.nix {
```
Comment (jtojnar) [inline pkgs/top-level/all-packages.nix]: Could you delete the file as well?

**Comment by Mic92** [comment]
cc @rnhmjoj regarding gnash

**Comment by Mic92** [comment]
I doubt that the tor-browser still uses `gst-ffmpeg`, because it is based on firefox. cc @joachifm

**Comment by pbogdan** [comment]
@Mic92 Yeah, I believe it should be safe to remove as gstreamer support seems to be gone (only looked into it today). I started working on a change that removes gstreamer dependencies from firefox and tor-browser expressions (FF doesn't use `gst-ffmpeg` but refers to some other legacy gstreamer packages).

**Comment by veprbl** [comment]
@pbogdan What is the status of this? What kind of testing did you do?

**Comment by Infinisil** [comment]
Nice work! Can you fix the merge conflict? Post a comment when done (GitHub doesn't send out notifications for new commits), I can then quickly test some things myself and merge it after that.

**Comment by pbogdan** [comment]
@Infinisil Thanks, conflicts resolved.

**Comment by Infinisil** [comment]
@GrahamcOfBorg build acoustidFingerprinter
@GrahamcOfBorg build cantata
@GrahamcOfBorg build renpy
@GrahamcOfBorg build ffmpegthumbnailer
@GrahamcOfBorg build ffms
@GrahamcOfBorg build xineLib
@GrahamcOfBorg build avxsynth
@GrahamcOfBorg build baresip
@GrahamcOfBorg build freerdp_legacy
@GrahamcOfBorg build guvcview

**Pull request merged**